go/src/github.com/mesosphere/rendler/scheduler/main.go

```diff
@@ -1,281 +1,284 @@
-				Value: proto.String(fmt.Sprintf("RENDLER-%d", taskId)),
-			},
-			SlaveId: offer.SlaveId,
-			Resources: []*mesos.Resource{
-				mesos.ScalarResource("cpus", TASK_CPUS),
-				mesos.ScalarResource("mem", TASK_MEM),
-			},
-		}
+		shutdown: make(chan interface{}),
+		done:     make(chan interface{}),
 	}
+	s.crawlQueue.PushBack(seedUrl)
+	return s
+}
 
-	makeCrawlTask := func(url string, offer mesos.Offer) *mesos.TaskInfo {
-		task := makeTaskPrototype(offer)
-		task.Name = proto.String("CRAWL_" + *task.TaskId.Value)
-		task.Executor = crawlExecutor
-		task.Data = []byte(url)
-		return task
+func (s *RendlerScheduler) makeTaskPrototype(offer *mesos.Offer) *mesos.TaskInfo {
+	taskId := s.tasksCreated
+	s.tasksCreated++
+	return &mesos.TaskInfo{
+		TaskId: &mesos.TaskID{
+			Value: proto.String(fmt.Sprintf("RENDLER-%d", taskId)),
+		},
+		SlaveId: offer.SlaveId,
+		Resources: []*mesos.Resource{
+			mesosutil.NewScalarResource("cpus", TASK_CPUS),
+			mesosutil.NewScalarResource("mem", TASK_MEM),
+		},
 	}
+}
 
-	makeRenderTask := func(url string, offer mesos.Offer) *mesos.TaskInfo {
-		task := makeTaskPrototype(offer)
-		task.Name = proto.String("RENDER_" + *task.TaskId.Value)
-		task.Executor = renderExecutor
-		task.Data = []byte(url)
-		return task
-	}
+func (s *RendlerScheduler) makeCrawlTask(url string, offer *mesos.Offer) *mesos.TaskInfo {
+	task := s.makeTaskPrototype(offer)
+	task.Name = proto.String("CRAWL_" + *task.TaskId.Value)
+	task.Executor = s.crawlExecutor
+	task.Data = []byte(url)
+	return task
+}
 
-	maxTasksForOffer := func(offer mesos.Offer) int {
-		// TODO(nnielsen): Parse offer resources.
-		count := 0
+func (s *RendlerScheduler) makeRenderTask(url string, offer *mesos.Offer) *mesos.TaskInfo {
+	task := s.makeTaskPrototype(offer)
+	task.Name = proto.String("RENDER_" + *task.TaskId.Value)
+	task.Executor = s.renderExecutor
+	task.Data = []byte(url)
+	return task
+}
 
-		var cpus float64 = 0
-		var mem float64 = 0
+func (s *RendlerScheduler) Registered(
+	driver sched.SchedulerDriver,
+	frameworkId *mesos.FrameworkID,
+	masterInfo *mesos.MasterInfo) {
+	log.Printf("Registered")
+}
 
-		for _, resource := range offer.Resources {
-			if resource.GetName() == "cpus" {
-				cpus = *resource.GetScalar().Value
-			}
+func (s *RendlerScheduler) Reregistered(driver sched.SchedulerDriver, masterInfo *mesos.MasterInfo) {
+	log.Println("Framework Re-Registered with Master ", masterInfo)
+}
 
-			if resource.GetName() == "mem" {
-				mem = *resource.GetScalar().Value
+func (s *RendlerScheduler) Disconnected(sched.SchedulerDriver) {
+	log.Println("Framework disconnected with Master")
+}
+
+func (s *RendlerScheduler) ResourceOffers(driver sched.SchedulerDriver, offers []*mesos.Offer) {
+	log.Printf("Received %d resource offers", len(offers))
+	for _, offer := range offers {
+		select {
+		case <-s.shutdown:
+			log.Println("Shutting down: declining offer on [", offer.Hostname, "]")
+			driver.DeclineOffer(offer.Id, DEFAULT_FILTER)
```

Comment:
I think you are right, but then it takes a while for the scheduler to shut down, so using `DeclineOffer` would allow the master to reuse those offers for other frameworks ASAP.